From 25612870f0ed00755afbb16d251aca0de5fb378e Mon Sep 17 00:00:00 2001
From: Juan RP <xtraeme@gmail.com>
Date: Mon, 26 Nov 2012 10:22:09 +0100
Subject: [PATCH] xbps-query: when checking revdeps from repos use the same
 pkgver from repo pkg.

This fixes showing revdeps for any pkg that uses dependencies like:

	foo>2.0<3.0
	blah<15
---
 bin/xbps-query/show-deps.c | 15 ++++++++++-----
 1 file changed, 10 insertions(+), 5 deletions(-)

diff --git a/bin/xbps-query/show-deps.c b/bin/xbps-query/show-deps.c
index f6f755f..2f39eae 100644
--- bin/xbps-query/show-deps.c
+++ bin/xbps-query/show-deps.c
@@ -127,12 +127,17 @@ repo_revdeps_cb(struct xbps_handle *xhp,
 int
 repo_show_pkg_revdeps(struct xbps_handle *xhp, const char *pkg)
 {
-	char *pattern;
+	prop_dictionary_t pkgd;
+	const char *pkgver;
 
 	if (xbps_pkg_version(pkg))
-		pattern = strdup(pkg);
-	else
-		pattern = xbps_xasprintf("%s-9999999", pkg);
+		pkgver = pkg;
+	else {
+		pkgd = xbps_rpool_find_pkg(xhp, pkg, false, false);
+		if (pkgd == NULL)
+			return ENOENT;
+		prop_dictionary_get_cstring_nocopy(pkgd, "pkgver", &pkgver);
+	}
 
-	return xbps_rpool_foreach(xhp, repo_revdeps_cb, pattern);
+	return xbps_rpool_foreach(xhp, repo_revdeps_cb, __UNCONST(pkgver));
 }
-- 
1.8.0

